packages/x-date-pickers/src/internals/components/PickersPopper.tsx

```diff
@@ -113,7 +113,7 @@ const PickersPopperPaper = styled(MuiPaper, {
 }>(({ ownerState }) => ({
   transformOrigin: 'top center',
   outline: 0,
-  ...(ownerState.placement === 'top' && {
+  ...(ownerState.placement?.includes('top') && {
```

Comment:
The `?` defensive logic looks wrong. It's impossible for the placement to be undefined.

```suggestion
  ...(ownerState.placement.includes('top') && {
```

Maybe this would fix it:

```diff
diff --git a/packages/x-date-pickers/src/internals/components/PickersPopper.tsx b/packages/x-date-pickers/src/internals/components/PickersPopper.tsx
index 8a7963525..3c9d67be5 100644
--- a/packages/x-date-pickers/src/internals/components/PickersPopper.tsx
+++ b/packages/x-date-pickers/src/internals/components/PickersPopper.tsx
@@ -54,7 +54,7 @@ export interface PickersPopperSlotsComponentsProps {
   desktopPaper?: SlotComponentProps<
     typeof MuiPaper,
     {},
-    PickerPopperProps & { placement: PopperPlacementType | undefined }
+    PickerPopperProps & { placement: PopperPlacementType }
   >;
   /**
```

It's impossible because if we provide undefined, it crashes in https://popper.js.org/docs/v2/constructors/#placement.

Reply:
@oliviertassinari Thank you for the suggestion!
Fixed it with https://github.com/mui/mui-x/pull/10069/commits/bb39de6a09de65e221c75b4a0b7a4302bb74b295
Also discovered an issue with this slot props and created a fix for it: https://github.com/mui/mui-x/pull/10103